engine/consensus/dkg/messaging_engine.go

```diff
@@ -104,40 +111,57 @@ func (e *MessagingEngine) Process(_ network.Channel, originID flow.Identifier, e
 func (e *MessagingEngine) process(originID flow.Identifier, event interface{}) error {
 	switch v := event.(type) {
 	case *msg.DKGMessage:
+		// messages are forwarded async rather than sync, because otherwise the message queue
+		// might get full when it's slow to process DKG messages synchronously and impact
+		// block rate.
+		e.forwardInboundMessageAsync(originID, v)
+		return nil
+	default:
+		return fmt.Errorf("invalid event type (%T)", event)
```

Comment:
```suggestion
		return fmt.Errorf("invalid event type (%T): %w", event, engine.IncompatibleInputTypeError)
```
return this error as a sentinel, which we can handle depending on the source of the input.